erpnext/selling/sales_common.js

```diff
@@ -632,5 +632,25 @@ erpnext.selling.SellingController = erpnext.TransactionController.extend({
 				}
 			})
 		}
-	}
+	},
+
+	add_set_cost_as_rate_button: function () {
```

Comment:
Check for "Restrict Amounts in Stock Reports to Role" permission before adding button